astropy/coordinates/sky_coordinate.py

```diff
@@ -206,12 +208,36 @@ class or the corresponding string alias.  The frame classes that are built in
     # info property.
     info = SkyCoordInfo()
 
+    # A counter to keep track of the (global) number of times SkyCoord has been
+    # initialized
+    _init_counter = 0
+
     def __init__(self, *args, copy=True, **kwargs):
 
         # these are frame attributes set on this SkyCoord but *not* a part of
         # the frame object this SkyCoord contains
         self._extra_frameattr_names = set()
 
+        # Increment the SkyCoord init counter:
+        SkyCoord._init_counter += 1
+
+        # If there is a threshold in the config, check the number of times
+        # SkyCoord has been initialized and possibly issue a warning
+        from . import conf
+        _counter_lim = conf.skycoord_init_counter_warn_threshold
+        if _counter_lim > 0 and SkyCoord._init_counter > _counter_lim:
+            url = "http://docs.astropy.org/en/latest/coordinates/skycoord.html#array-operations"
+            warnings.warn("We noticed that you have initialized SkyCoord more "
+                          f"than {_counter_lim} times. It is generally much "
+                          "more efficient to instead create a single SkyCoord "
+                          "instance with array-valued coordinate components, "
+                          f"as demonstrated in this documentation page: {url} "
+                          "However, if you do need to initialize many SkyCoord "
+                          "instances and would like to remove this check, set "
+                          "the astropy.coordinates configuration item "
+                          "'skycoord_init_counter_warn_threshold=0'",
```

Comment:
Better to give the full story, `coordinates.conf.skycoord_init_counter_warn_threshold = 0`.  Most users probably won't know offhand how to set a configuration item.  I don't remember now if the config items need to be typed, since setting this to `None` for no threshold would be more natural.

Reply:
Alternatively, explain the whole thing like:
```suggestion
                          "'from astropy.coordinates import conf",
                          "conf.skycoord_init_counter_warn_threshold=0'"
```